python/tests/res/enkf/test_site_config.py

```diff
@@ -90,4 +91,38 @@ def test_constructors(self):
             self.assertEqual(site_config_dict, site_config_user_file)
 
             with self.assertRaises(ValueError):
-                site_config = SiteConfig(user_config_file=config_file, config_dict=snake_config_dict)
\ No newline at end of file
+                site_config = SiteConfig(user_config_file=config_file, config_dict=snake_config_dict)
+
+    @tmpdir()
+    def test_site_config_hook_workflow(self):
+        site_config_filename = "test_site_config"
+        test_config_filename = "test_config"
+        site_config_content = """
+LOAD_WORKFLOW_JOB ECHO_WORKFLOW_JOB
+LOAD_WORKFLOW ECHO_WORKFLOW
+HOOK_WORKFLOW ECHO_WORKFLOW PRE_SIMULATION
+"""
+
+        with open(site_config_filename, "w") as fh:
+            fh.write(site_config_content)
+
+        with open(test_config_filename, "w") as fh:
+            fh.write("NUM_REALIZATIONS 1\n")
+
+        with open("ECHO_WORKFLOW_JOB", "w") as fh:
+            fh.write(
+                """INTERNAL False
+EXECUTABLE echo
+MIN_ARG 1
+"""
+            )
+
+        with open("ECHO_WORKFLOW", "w") as fh:
+            fh.write("ECHO_WORKFLOW_JOB hello")
+
+        os.environ["ERT_SITE_CONFIG"] = site_config_filename
```

Comment:
I suggest that we use the `monkeypatch` approach @oyvindeide introduced some time back instead: https://github.com/equinor/libres/blob/b2801739be29a652e733f2ae239ea0ec25198683/python/tests/res/fm/test_rms_config.py

Reply:
:+1: